lib/rubocop/cop/solidus/spree_gateway_bogus_deprecated.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Solidus
+      # This cop finds Spree::Gateway::Bogus calls and replaces them with the Spree::PaymentMethod::BogusCreditCard call
+      # This cop is needed as the Spree::Gateway::Bogus has been deprecated in future version.
+      #
+      # @example EnforcedStyle:
+      #   # bad
+      #   Spree::Gateway::Bogus.new
+      #
+      #   # good
+      #   Spree::PaymentMethod::BogusCreditCard.new
+      #
+      #   # bad
+      #   Spree::Gateway::Bogus.create
+      #
+      #   # good
+      #   Spree::PaymentMethod::BogusCreditCard.create
+      #
+      #   # bad
+      #   Spree::Gateway::Bogus.create!
+      #
+      #   # good
+      #   Spree::PaymentMethod::BogusCreditCard.create!
+      #
+      #
+      class SpreeGatewayBogusDeprecated < Base
+        extend AutoCorrector
+        MSG = 'Spree::Gateway::Bogus is deprecated. Please use Spree::PaymentMethod::BogusCreditCard instead.'
+
+        # TODO: Don't call `on_send` unless the method name is in this list
+        # If you don't need `on_send` in the cop you created, remove it.
+        RESTRICT_ON_SEND = %i[new create create!].freeze
```

Comment:
This works for `Spree::Gateway::Bogus.new`, `Spree::Gateway::Bogus.create`, `Spree::Gateway::Bogus.create!`

But this will probably not work in case of `Spree::Gateway::Bogus.update`, `Spree::Gateway::Bogus.dup`, `Spree::Gateway::Bogus.find_by`, `Spree::PaymentMethod::BogusCreditCard.where`

You can add these cases mentioned above to the spec file and try your cop on them